In [1]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.neighbors import KNeighborsClassifier

In [2]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

In [3]:
df = pd.read_csv(r'processed_data.csv')

In [4]:
df.head()

Unnamed: 0       age    fnlwgt  capital-gain  capital-loss  hours-per-week  \
0           0 -0.995129  0.351675     -0.144804     -0.217127       -0.034087   
1           1 -0.046942 -0.945524     -0.144804     -0.217127        0.772930   
2           2 -0.776316  1.394723     -0.144804     -0.217127       -0.034087   
3           3  0.390683 -0.277844      0.886874     -0.217127       -0.034087   
4           4 -1.505691 -0.815954     -0.144804     -0.217127       -0.841104   

   workclass_1  workclass_2  workclass_3  workclass_4  ...  native-country_32  \
0            0            0            1            0  ...                  0   
1            0            0            1            0  ...                  0   
2            1            0            0            0  ...                  0   
3            0            0            1            0  ...                  0   
4            0            0            1            0  ...                  0   

   native-country_33  native-country_34  native-country_35  native-country_36  \
0                  0                  0                  0                  0   
1                  0                  0                  0                  0   
2                  0                  0                  0                  0   
3                  0                  0                  0                  0   
4                  0                  0                  0                  0   

   native-country_37  native-country_38  native-country_39  native-country_40  \
0                  0                  1                  0                  0   
1                  0                  1                  0                  0   
2                  0                  1                  0                  0   
3                  0                  1                  0                  0   
4                  0                  1                  0                  0   

   income_1  
0         0  
1         0  
2         1  
3         1  
4         0  

[5 rows x 113 columns]

In [5]:
x = df.iloc[:,:-1].values
y = df.iloc[:,-1].values

In [6]:
xtrain,xtest,ytrain,ytest = train_test_split(x,y,test_size=0.2,random_state=0)

In [7]:
lr = LogisticRegression()

In [8]:
lr.fit(xtrain,ytrain)

LogisticRegression()

In [9]:
lr_pred = lr.predict(xtest)

In [10]:
accuracy_score(ytest,lr_pred)

0.7972156822602109

In [11]:
print('---------------------------------------------------------------------------------------------------------------------------------------')

---------------------------------------------------------------------------------------------------------------------------------------


In [12]:
svc = SVC(kernel='rbf')

In [13]:
svc.fit(xtrain,ytrain)

SVC()

In [14]:
svc_pred = svc.predict(xtest)

In [15]:
accuracy_score(ytest,svc_pred)

0.7595455010748285

In [16]:
print('---------------------------------------------------------------------------------------------------------------------------------------')

---------------------------------------------------------------------------------------------------------------------------------------


In [17]:
rf = RandomForestClassifier()

In [18]:
rf.fit(xtrain,ytrain)

RandomForestClassifier()

In [19]:
rf_pred = rf.predict(xtest)
accuracy_score(ytest,rf_pred)

0.8566895280990889

In [20]:
print('---------------------------------------------------------------------------------------------------------------------------------------')

---------------------------------------------------------------------------------------------------------------------------------------


In [21]:
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(xtrain,ytrain)
knn_pred = knn.predict(xtest)
accuracy_score(ytest,knn_pred)

0.7253557170641827

### Obesrvation
Random Forrest has the highest accuracy i.e 85.67%

### Conclusion
Hyperparameter Tuning can be used to further improve the model.